src/qp/optimizer/RandomInitialPlan.java

```diff
@@ -195,10 +196,39 @@ public void createProjectOp() {
         }
     }
 
+    public void createGroupbyOp() {
+        Operator base = root;
+        if (!groupbylist.isEmpty()) {
+            // Attribute in SELECT clause must appear in Group By clause or is a primary key.
+            for (Attribute attr : projectlist) {
+                if (!(attr.isPrimaryKey() || groupbylist.contains(attr))) {
+                    System.out.println("Attribute does not appear in Group By clause and is not a primary key.");
+                    System.exit(1);
+                }
+            }
+            root = new Groupby(base, groupbylist, OpType.GROUPBY);
+            root.setSchema(base.getSchema());
+        }
+    }
+
     public void createOrderbyOp() {
         Operator base = root;
         if (!orderbylist.isEmpty()) {
-            root = new Orderby(base, orderbylist, OpType.ORDERBY, isDesc);
+            if (!groupbylist.isEmpty()) {
+                // Combines groupby list and orderby list to maintain the groups when sorting.
+                ArrayList<Attribute> combinedList = new ArrayList<>(groupbylist);
```

Comment:
does this combination lead to the expected results? For a query that has both groupby and orderby, is this implied that these operations can be composed?